capitals.py

```diff
@@ -1,15 +1,23 @@
-def letters_count():
-    print('Please write your text')
-    input_text = input()
-    count_lower = 0
+def count_letters(string):
     count_upper = 0
+    count_lower = 0
 
-    for i in input_text:
+    for i in string:
         if (i.islower()):
             count_lower = count_lower + 1 
         elif (i.isupper()):
             count_upper = count_upper + 1
-    print('There are {} uppercase letters'.format(count_upper))
-    print('There are {} lowercase letters'.format.(count_lower))
 
-letters_count()
+    return (count_upper, count_lower)
+
+def main():
+    print('Please write your text')
+    input_text = input()
+
+    results = count_letters(input_text)
+    
+    print('There are {} uppercase letters'.format(results[0]))
+    print('There are {} lowercase letters'.format(results[1]))
+
+if __name__ == "__main__":
```

Comment:
am citit acum despre de ce se foloseste __name__ 
nu prea inteleg utilitatea lui, sau scenariile in care acesta e necesar. daca corect am inteles. __name__ e o variabila speciala, definita de python inainte de rularea programului si daca fisierul dat se executa ca program principal (main), atunci el primeste argumentul __main__ 
respectiv daca din if, daca fisierul dat e chiar cel cu programul principal el va merge sa execute functia main. Daca nu, nu se va rula nimic? la ce ajuta asta?

Reply:
Uite bucata asta am explicat în comentariu mai sus. Utilitatea este că la import în alt fișier poate nu ai nevoie să fie executat automat main sau altă logică aferentă. Asta dă posibilitatea să folosești același cod în 2 moduri: ca program executabil; și ca logică reutilizabilă prin import. 

Spre exemplu ai o aplicație pentru conversia de valută. Scrii în consolă ceva de gen `$currconv -i USD -o MDL 45.35`. Și la un moment dat înșelegi că vrei să folosești acest program într-o aplicație online magazin (spre exemplu) pu a converti prețurile când omul alege altă valută. Și pentru nu a scri din nou logica de conversie care deja o ai tu ai să imporți acest cod ca modul. Dar nu ai nevoie de logica de prelucrare a parametrilor -i, -o etc. ai nevoie doar de funcție unde indici ce în ce și cât. Și tot ce trebuie să faci în așa caz e să imporți acest fișier. Atât.